sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyClientBuilder.java

```diff
@@ -155,7 +156,9 @@ public KeyClientBuilder() {
      * and {@link #retryPolicy(RetryPolicy)} have been set.
      */
     public KeyClient buildClient() {
-        return new KeyClient(buildAsyncClient());
+        KeyAsyncClient asyncClient = buildAsyncClient();
+        KeyServiceVersion serviceVersion = version != null ? version : KeyServiceVersion.getLatest();
+        return new KeyClient(asyncClient.getVaultUrl(), asyncClient.getHttpPipeline(), serviceVersion);
```

Comment:
is it possible to avoid KeyAsyncClient  creation here?
I'd assume that what we need is the impl client? See

https://github.com/Azure/azure-sdk-for-java/blob/3c92d218de218b2fe8b0bcc39e2a653f4f1a59d1/sdk/anomalydetector/azure-ai-anomalydetector/src/main/java/com/azure/ai/anomalydetector/AnomalyDetectorClientBuilder.java#L239-L250